src/libs/panel_resizer.js

```diff
@@ -0,0 +1,155 @@
+import PanelResizerActionsView from '../views/panel_resizer_actions.dot';
+
+const REDUCED_PANEL_CLASS = 'reduced';
+const MAXIMIZED_PANEL_CLASS = 'full';
+
+function getClosest(arr, goal) {
+  return arr.reduce((prev, curr) =>
+    Math.abs(curr - goal) < Math.abs(prev - goal) ? curr : prev
+  );
+}
+
+export default class PanelResizer {
+  constructor(panel) {
+    this.panel = panel;
+    this.timer = null;
+    this.holding = false;
+    this.isTransitioning = false;
+    this.reduced = false;
+    this.maximized = false;
+
+    this.moveCallback = e => this.move(e);
+  }
+
+  get RESIZABLE_PANEL_CLASS() {
+    return 'resizable_panel';
+  }
+
+  get HANDLE_CLASS() {
+    return 'resizable_panel_handle';
+  }
+
+  get resizableElement() {
+    return document.querySelector(`[data-cid="${this.panel.cid}"] .${this.RESIZABLE_PANEL_CLASS}`);
+  }
+
+  get handleElement() {
+    return document.querySelector(`[data-cid="${this.panel.cid}"] .${this.HANDLE_CLASS}`);
+  }
+
+  /**
+   * Triggered on mouse down of the panel resizer
+   * @param {MouseEvent|TouchEvent} e event
+   */
+  async holdResizer({event}) {
```

Comment:
It doesn't seem to be an `async` method.

Reply:
`async` is now removed